articles/azure-app-configuration/enable-dynamic-configuration-dotnet-core-push-refresh.md

```diff
@@ -175,6 +175,49 @@ A random delay is added before the cached value is marked as dirty to reduce pot
 > [!NOTE]
 > To reduce the number of requests to App Configuration when using push refresh, it is important to call `SetCacheExpiration(TimeSpan cacheExpiration)` with an appropriate value of `cacheExpiration` parameter. This controls the cache expiration time for pull refresh and can be used as a safety net in case there is an issue with the Event subscription or the Service Bus subscription. The recommended value is `TimeSpan.FromDays(30)`.
 
+## Optional: Incorporate Synchronization tokens for real-time consistency
+
+> [!NOTE]
+> Support for using synchronization-tokens in push refresh workflow is available in .NET configuration provider version 5.0.0-preview (or later). 
+
+Due to the nature of distributed systems, real-time consistency between requests is difficult to enforce implicitly. We use synchronization tokens to address this issue. The Event Grid event contains the sync-token that can be utilized to guarantee real-time consistency between different client instances and requests.
+
+To use a `sync-token` in your push refresh workflow, make the following changes to your **Program.cs** file:
+
+1. Add a dependency on Azure.messaging.EventGrid Package:
+```cs
+using Azure.Messaging.EventGrid;
+```
+2. Make the following changes to the RegisterRefreshEventHandler method:
+```cs
+ private static void RegisterRefreshEventHandler()
+        {
+            string serviceBusConnectionString = Environment.GetEnvironmentVariable(ServiceBusConnectionStringEnvVarName);
+            string serviceBusTopic = Environment.GetEnvironmentVariable(ServiceBusTopicEnvVarName);
+            string serviceBusSubscription = Environment.GetEnvironmentVariable(ServiceBusSubscriptionEnvVarName);
+            SubscriptionClient serviceBusClient = new SubscriptionClient(serviceBusConnectionString, serviceBusTopic, serviceBusSubscription);
+            serviceBusClient.RegisterMessageHandler(
+                handler: (message, cancellationToken) =>
+                {
+                    EventGridEvent eventGridEvent = EventGridEvent.Parse(BinaryData.FromBytes(message.Body));
+
+                    eventGridEvent.TryCreatePushNotification(out PushNotification pushNotification);
+
+                    _refresher.ProcessPushNotification(pushNotification);
+
+                    return Task.CompletedTask;
+                },
+                exceptionReceivedHandler: (exceptionargs) =>
+                {
+                    Console.WriteLine($"{exceptionargs.Exception}");
+                    return Task.CompletedTask;
+                });
+        }
+```
+
```

Comment:
Now that I read it again, I feel like we need a little more explanation before we talk about SetDirty.

How about this:
"The `TryCreatePushNotification` method extracts the synchronization tokens and other relevant information from an `EventGridEvent` object and creates a `PushNotification` object. The `ProcessPushNotification` method then prepares the client to use these synchronization tokens in the next request to App Configuration service, thereby guaranteeing real-time consistency for any subsequent request."